I have a couple thoughts:

- PNNL CI for hiop is still on the exasgd allocation, which will need to change or be dropped as that account is no longer funded.
- there are some weird errors in the log - could be /tmp issues I was running into on the dl partition from the install of cuda12.3, might have to ping Tim about cleaning up /tmp on the dmi nodes

```
/tmp/slurmd/job3687796/slurm_script: line 107: module: command not found
ModuleCmd_Use.c(231):ERROR:64: Directory '/share/apps/modules/tools' not found
ModuleCmd_Use.c(231):ERROR:64: Directory '/share/apps/modules/compilers' not found
ModuleCmd_Use.c(231):ERROR:64: Directory '/share/apps/modules/mpi' not found
Using ./scripts/inclineVariables.sh
Currently Loaded Modulefiles:
  1) gcc/8.4.0         4) raja/0.14.0       7) magma/2.6.1
  2) rocm/4.5.1        5) openblas/0.3.18
  3) umpire/6.0.0      6) cmake/3.19.6
Build step
~/gitlab/142740/build ~/gitlab/142740
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.
loading initial cache file /people/svcexasgd/gitlab/142740/scripts/clang-hip.cmake
CMake Error: The source directory "/people/svcexasgd/gitlab/142740/build" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
./scripts/defaultBuild.sh: line 16: -Wl,-rpath,/share/apps/gcc/8.4.0/lib64: No such file or directory
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -228,6 +228,7 @@ if(HIOP_USE_GPU)
   endif()
 
   if(HIOP_USE_CUDA)
+    set(HIOP_USE_RAJA "ON") #CUDA/HIP requires using RAJA.
     enable_language(CUDA)
     check_language(CUDA)
 
@@ -269,7 +270,6 @@ if(HIOP_USE_GPU)
         "\n* HIOP_USE_HIP"
     )
   endif()
-  set(HIOP_USE_RAJA "ON") #CUDA/HIP requires using RAJA. This dependency will be removed in a future release, after somee necessary implementations.
 endif(HIOP_USE_GPU)
 
 if(HIOP_USE_RAJA)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -269,6 +269,7 @@ if(HIOP_USE_GPU)
         "\n* HIOP_USE_HIP"
     )
   endif()
+  set(HIOP_USE_RAJA "ON") #CUDA/HIP requires using RAJA. This dependency will be removed in a future release, after somee necessary implementations.
 endif(HIOP_USE_GPU)
 
 if(HIOP_USE_RAJA)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -227,8 +227,13 @@ if(HIOP_USE_GPU)
     set(HIOP_MAGMA_DIR CACHE PATH "Path to Magma directory")
   endif()
 
-  if(HIOP_USE_CUDA)
+  if(NOT HIOP_USE_RAJA)
     set(HIOP_USE_RAJA "ON") #CUDA/HIP requires using RAJA.
+    message(STATUS "Enable RAJA since HIOP_USE_GPU is ON. HiOp requires using RAJA with CUDA/HIP.")
+  endif(NOT HIOP_USE_RAJA)
+
+
+  if(HIOP_USE_CUDA)
     enable_language(CUDA)
     check_language(CUDA)
 

--- a/src/LinAlg/LinAlgFactory.cpp
+++ b/src/LinAlg/LinAlgFactory.cpp
@@ -353,7 +353,7 @@ hiopMatrixDense* LinearAlgebraFactory::create_matrix_dense(const ExecSpaceInfo& 
       return nullptr;
 #endif //HIOP_USE_RAJA
     } else {// for if(hi.exec_backend_ == "RAJA")
-
+      assert(false && "device memory backend not available because HiOp was not built with RAJA");
       if(mem_space_upper == "CUDA") {
 #ifdef HIOP_USE_CUDA
           assert(mem_space_upper == "DEVICE");

